### PR TITLE
feat: exportable ControlTransformer

### PR DESCRIPTION
**Pull request by egorchakov**
This PR makes `ControlTransformer` with a single `PolicyObjective` `torch.export.export`able.

Vanilla export didn't work seemingly due to `TensorDict` and `optree` APIs -- replaced them with `torch.utils._pytree`. In order to avoid too much export vs non-export path code duplication, some modules (most notably `ModuleDict`) now operate on vanilla `dicts` rather than `TensorDicts`.

Additionally, exported `.forward` methods can't take `Module` arguments, which is how we used to pass the (shared) encoder to objectives. Objectives now take encoders in `__init__`, shared in YAML with [`hydra-once`](https://github.com/swamidass/hydra-once).

Tested with `tests/test_export.py` (module -> export -> AOT Inductor) and a training smoke test. Some other functionality (e.g. finetuning, prediction) might be broken as the tests don't hit all code paths -- will fix problems as they appear.

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -57,12 +56,16 @@ predict +ARGS: generate-config
...
 predict-debug +ARGS: generate-config
     uv run src/rmind/scripts/predict.py \
         --config-path {{ justfile_directory() }}/config \
         --config-name predict.yaml {{ ARGS }}
 
+@export *ARGS: generate-config
+    uv run src/rmind/scripts/export.py \
+        --config-path {{ justfile_directory() }}/config \
+        --config-name export.yaml {{ ARGS }}
+
```
Comment (koritsky) [inline justfile:68]: i cant find `src/rmind/scripts/export.py` and corresponding `export.yaml` file

Comment (egorchakov) [reply justfile:68]: ah good catch, this was from the initial experiments; superseded by tests

File: src/rmind/components/objectives/policy.py [hunk 2]
```diff
@@ -37,20 +45,113 @@ class PolicyObjective(Objective):
...
+                Modality.SPECIAL,
+                SpecialToken.OBSERVATION_HISTORY,
+            ))
+
+            observation_summary = embeddings.get((
+                Modality.SPECIAL,
+                SpecialToken.OBSERVATION_SUMMARY,
+            ))
+
+            waypoints = episode.input_embeddings[Modality.CONTEXT, "waypoints"][
```
Comment (sandhawalia) [inline src/rmind/components/objectives/policy.py:94]: encoder output ? @koritsky

File: tests/test_export.py [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+def test_module_export_aoti(
+    module: Module, args: tuple[Any], args_export: tuple[Any]
+) -> None:
+    module = module.eval()
+    with torch.inference_mode():
+        module_output = module(*args)
+        exported = torch.export.export(module, args=args_export)
+        exported_output = exported.module()(*args_export)
+
+    assert all(
```
Comment (sandhawalia) [inline tests/test_export.py:311]: ❤️

**Comment by sandhawalia** [comment]
proposed changes
- waypoints from encoder output
- PT checkpoint be loadable for FT
- run a FT 
- build an exportable artifact

File: src/rmind/components/objectives/policy.py (hunk 2 above)
Comment (koritsky) [reply src/rmind/components/objectives/policy.py:94]: https://github.com/yaak-ai/rmind/tree/feat/export-suggesh
@egorchakov pls consider merging

File: src/rmind/components/objectives/policy.py [hunk 4]
```diff
@@ -37,20 +45,113 @@ class PolicyObjective(Objective):
...
+            ]
+
+            observation_history = embedding[
+                :,
+                episode.index[Modality.SPECIAL.value][  # pyright: ignore[reportArgumentType]
+                    SpecialToken.OBSERVATION_HISTORY.value
+                ][-1],
+            ]
+
+            waypoints = episode.input_embeddings[Modality.CONTEXT.value]["waypoints"][  # pyright: ignore[reportArgumentType]
```
[the remark this answers is not in the document]
Comment (koritsky) [reply src/rmind/components/objectives/policy.py:113]: danone

**Review by koritsky — APPROVED** [review]

File: src/rmind/components/mask.py [hunk 5]
```diff
@@ -24,7 +24,7 @@ def __new__(metacls, cls: str, bases: tuple[type, ...], classdict: _EnumDict):
         return super().__new__(metacls, cls, bases, classdict)
 
 
-class XFormersAttentionMaskLegend(float, Enum, metaclass=AttentionMaskLegend):
+class TorchAttentionMaskLegend(float, Enum, metaclass=AttentionMaskLegend):
```
Comment (sandhawalia) [inline src/rmind/components/mask.py:27]: 🫡

**Review by sandhawalia — CHANGES_REQUESTED** [review]
would like raw.yaml to reverted back for readability.

**Review by sandhawalia — APPROVED** [review]
🔥 there is one change needed on main. To fix an issue we found with RHMC and indexing with tensordict. I'd say we fix that first and then rebase this PR and merge

**Review by koritsky — CHANGES_REQUESTED** [review]

**Comment by koritsky** [comment]
> training run and then we goood ?

https://wandb.ai/yaak/cargpt/runs/smb3nggp/overview
